### DISCUSSION TITLE
Circles in `sf`

### DISCUSSION THREAD

**Discussion by roaldarbol** [r-spatial/sf#2444] [General]
Hi there! I'm just beginning to get a hold of what is possible with `sf`, but am super excited so far!

One thing I am trying to figure out what is possible with `sf` in relation to circles. So a few questions (and I'm happy to create issues and/or PRs where it would make sense afterwards) in case anyone has any knowledge and would like to chat about it:
- Is it possible to fit a circle to a set of points (which do not themselves make a perfect circle), similar to `st_ bbox`? I'm imagining three possible methods/modes would be possible: `inside`, which makes the largest circle that fit *inside* all the points; `outside` which is the smallest circle that fit outside all of the points; `mean` which minimises the mean of all the distances (as in https://r-spatial.github.io/sf/reference/st_nearest_feature.html). `outside` would be the same as `st_bbox`.
- Making zones should be relatively easy I think, with `st_ buffer` from the central point. 
	- Maybe there would be a way to divide a circle up into "slices (of pie)" as an alternative set of zones?
	
Thanks, and looking forward to feedback!

**Comment by edzer** [comment]
The function is there in GEOS, `GEOSMinimumBoundingCircle_r`, but we don't have it interfaced in `sf`. Implementing it in `sf` is reasonably trivial, but involves conditioning on the GEOS version in which it appeared.

**Reply by roaldarbol** [reply]
Awesome! Seems to have been implemented in  3.9.1 according to the docs of the R package of the same name (in which it was implemented in 0.1.0) https://paleolimbot.github.io/geos/news/index.html.

**Reply by roaldarbol** [reply]
@edzer Should I open an issue for this to keep it on the radar?

**Reply by edzer** [reply]
That's OK.
